desktop/src/navigator.rs

```diff
@@ -172,12 +174,32 @@ impl NavigatorBackend for ExternalNavigatorBackend {
                     )));
                 }
 
-                let mut buffer = vec![];
+                let url = if let Some(uri) = response.effective_uri() {
+                    // There seems no convenient way of formatting a `Uri`.
+                    // Build up a string manually.
+                    let mut url = String::new();
+                    if let Some(scheme) = uri.scheme_str() {
+                        url.push_str(scheme);
+                    }
+                    url.push_str("://");
+                    if let Some(authority) = uri.authority() {
+                        url.push_str(authority.as_str());
+                    }
+                    if let Some(path_and_query) = uri.path_and_query() {
+                        url.push_str(path_and_query.as_str());
+                    }
+                    url
+                } else {
```

Comment:
Looks like [`Uri` implements `Display`](https://docs.rs/http/0.2.7/http/uri/struct.Uri.html#impl-Display) -- does `uri.to_string()` work?

Reply:
Missed that. Thanks!